vald/start_test.go

```diff
@@ -23,7 +26,9 @@ func TestPanic(t *testing.T) {
 			timer = time.AfterFunc(1*time.Millisecond, func() {
 				timeoutCancel()
 			})
+			blocksSeenMutex.Lock()
```

Comment:
Is this really how mutexes are supposed to be used in go? Manual locking and unlocking? If so, might be cleaner to just use an atomic integer for blockSeen